### DISCUSSION TITLE
Are themes for multiple purposes considered?

### DISCUSSION THREAD

**Discussion by devenliu** [OrchardCMS/OrchardCore#14403] [General]
### Is your feature request related to a problem? Please describe.

Currently we can only have "Admin" and "Site" themes, but sometimes we need themes for more purposes, such as workbenches, workspaces, and so on.

### Describe the solution you'd like

Hopefully, themes for more scenarios can be supported in some way, such as:

1. Add fields to the Manifest.cs file of the theme module.
2. Provide programmatic custom implementations like AdminTheme and SiteTheme.

Or something better

**Comment by hishamco** [comment]
>  themes, but sometimes we need themes for more purposes, such as workbenches, workspaces, and so on.

In the end will fall into the site or admin theme, right?

> Hopefully, themes for more scenarios can be supported in some way, such as:

> Add fields to the Manifest.cs file of the theme module.
> Provide programmatic custom implementations like AdminTheme and SiteTheme.

Could you please elaborate?

**Comment by Piedone** [comment]
While I'm also asking you to elaborate a bit like Hisham, are you aware that by implementing an [`IThemeSelector`](https://github.com/OrchardCMS/OrchardCore/blob/7cb1cd2dc69ff5c5305e9feee80ca59d43d793b4/src/OrchardCore/OrchardCore.DisplayManagement/Theming/IThemeSelector.cs#L10), you can add any custom logic to determine what the currently applied theme should be at any given time?

**Comment by devenliu** [comment]
Well, what I want to express is that we use OC to build a SaaS website.

The website theme is perfect for the front end of the website, no problem.

Admin themes are great for developers, but often we need to provide customers with a similar admin theme, but the UI will be easier to use, and it would be too onerous to completely rewrite the default Admin theme.

So we want to add a fully customizable theme for more uses that customers can choose like site theme and admin theme.

**Comment by jtkech** [comment]
Just for info you can have a custom Theme whose manifest file sepcifies for example TheAdmin as a `BaseTheme`. This way you inherit of all the shape views from TheAdmin and then add in your custom derived theme only the shape views you want to override.

**Comment by sebastienros** [comment]
> but often we need to provide customers with a similar admin theme

Then I'd say that's part of your front-end, with front-end auth, and custom css. Even though it's the admin part for your users, that not the admin part of orchard. Your front end can be a "workbench" or anything you want.
